base/acme/src/main/java/org/dogtagpki/acme/server/ACMEChallengeProcessor.java

```diff
@@ -95,11 +96,11 @@ public void finalizeValidAuthorization() throws Exception {
         // If the final state is "valid", then the server MUST include an "expires"
         // field.
 
-        // set valid authorization to expire in 30 minutes
-        // TODO: make it configurable
-
-        Date expirationTime = new Date(currentTime + 30 * 60 * 1000);
-        authorization.setExpirationTime(expirationTime);
+        ACMEValidity validity = engine.getConfig().getValidity("validAuthorization");
```

Comment:
As I mention in comments below, the implementation should be such that there is always a valid value for the TTL of an object, and we set the expiration field unconditionally.

I do not think we should have code paths that leave expiration fields unset - even if this would only happen in the event of misconfiguration.

Reply:
It has a default value now.